pyproject.toml

```diff
@@ -33,18 +33,18 @@ dependencies = [
     "numpy>=1.24",
     "pandas>=2.0",
     "xarray>=2023.04",
-    "netCDF4",
+    "netCDF4>=1.7",
```

Comment:
We discussed making NetCDF an optional dependency before in https://github.com/GenericMappingTools/pygmt/pull/429. Maybe we could revisit that discussion?

Reply:
That's a good point. NetCDF4 was needed by xarray when reading and writing netCDF files. As mentioned in https://github.com/GenericMappingTools/pygmt/pull/239#issuecomment-438449187, previously, our `load_xxx` functions (e.g., `load_earth_relief`) called `xarray.load_dataarray` to load local netCDF grids into an xarray.DataArray object, making netCDF a highly "required" dependency.

After we refactor our `load_xxx` function in https://github.com/GenericMappingTools/pygmt/pull/3120, we no longer call `xarray.load_dataarray`, so netCDF should be no longer highly needed by PyGMT.

I've opened PR #3643 to see how the PyGMT relies on the netCDF package.